parser/src/lib.rs

```diff
@@ -303,6 +312,68 @@ pub fn parse<'f>(format_string: &'f str) -> Result<Vec<Fragment<'f>>, Cow<'stati
     Ok(fragments)
 }
 
+impl Level {
+    pub fn as_str(self) -> &'static str {
+        match self {
+            Level::Trace => "trace",
+            Level::Debug => "debug",
+            Level::Info => "info",
+            Level::Warn => "warn",
+            Level::Error => "error",
+        }
+    }
+
+    // returns a list of features of which one has to be enabled for this Level to be active
+    pub fn necessary_features(self, debug_assertions: bool) -> &'static [&'static str] {
```

Comment:
This should maybe go into the macro crate, since it's only needed there

Reply:
if I understand it correctly I can't do that because that would spread the impl across different crates which is not allowed?

Reply:
Yeah, it would have to be a free function

Reply:
welp, that makes sense. done